src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -498,5 +498,75 @@ private static bool ShouldOptimizeOutInitializer(BoundStatement initializer)
             BoundExpression rhs = assignment.Right;
             return rhs.IsDefaultValue();
         }
+
+        /// <summary>
+        /// Receivers of struct methods are alowed to be RValues or LValues.
+        /// Whether the mutations from the method are propagated back to the 
+        /// receiver instance is conditional on whether the receiver is an LValue.
+        /// 
+        /// An inconvenient situation may arise when an the receiver is an RValue expression,
+        /// which is trivially reduced during lowering to one of its operands and 
+        /// such operand happens to be an LValue. That operation alone would 
+        /// expose the operand to mutations while it would not be exposed otherwise.
+        /// I.E. the transformation becomes semantically observable.
+        /// 
+        /// To prevent such situations, we will wrap the operand into a node whose only 
+        /// purpose is to never be an LValue.
+        /// </summary>
+        private static BoundExpression EnsureNotLvalueReceiver(BoundExpression expr)
```

Comment:
Correct term would be "Assignable Variable".  

LValue is not used very consistently in the codebase, sometimes it also means "assignable", sometimes it does not.

Will change to something less confusing.

---
In reply to: [108197238](https://github.com/dotnet/roslyn/pull/18166#discussion_r108197238) [](ancestors = 108197238)